pkg/registry/pod/etcd/etcd.go

```diff
@@ -183,14 +189,23 @@ func (r *BindingREST) setPodHostTo(ctx api.Context, podID, oldMachine, machine s
 }
 
 // assignPod assigns the given pod to the given machine.
-func (r *BindingREST) assignPod(ctx api.Context, podID string, machine string) error {
-	finalPod, err := r.setPodHostTo(ctx, podID, "", machine)
-	if err != nil {
+func (r *BindingREST) assignPod(ctx api.Context, podID string, machine string, annotations map[string]string) error {
+	translateError := func(err error) error {
+		err = etcderr.InterpretCreateError(err, "binding", "")
+		err = etcderr.InterpretGetError(err, "pod", podID)
+		err = etcderr.InterpretUpdateError(err, "pod", podID)
+		if _, ok := err.(*errors.StatusError); !ok && err != nil {
+			err = errors.NewConflict("binding", "", err)
```

Comment:
I'm concerned about returning conflict for constraint failures. Clients may retry assuming it's a resourceVersion mismatch.